Essentials/src/com/earth2me/essentials/commands/Commandeco.java

```diff
@@ -32,82 +29,61 @@ public void run(final Server server, final CommandSource sender, final String co
-            } catch (MaxMoneyException ex) {
-                // Take shouldn't be able to throw a max money exception
+            switch (cmd) {
+                case GIVE: {
+                    player.giveMoney(userAmount, sender, UserBalanceUpdateEvent.Cause.COMMAND_ECO);
+                    break;
+                }
+                case TAKE: {
+                    if (player.getMoney().subtract(userAmount).compareTo(ess.getSettings().getMinMoney()) >= 0) {
+                        player.takeMoney(userAmount, sender, UserBalanceUpdateEvent.Cause.COMMAND_ECO);
+                    } else {
+                        ess.showError(sender, new Exception(tl("minimumBalanceError", NumberUtil.displayCurrency(ess.getSettings().getMinMoney(), ess))), commandLabel);
+                    }
+                    break;
+                }
+                case RESET:
+                case SET: {
+                    BigDecimal minBal = ess.getSettings().getMinMoney();
+                    BigDecimal maxBal = ess.getSettings().getMaxMoney();
+                    boolean underMin = (userAmount.compareTo(minBal) < 0);
+                    boolean aboveMax = (userAmount.compareTo(maxBal) > 0);
+                    player.setMoney(underMin ? minBal : aboveMax ? maxBal : userAmount, UserBalanceUpdateEvent.Cause.COMMAND_ECO);
+                    player.sendMessage(tl("setBal", NumberUtil.displayCurrency(player.getMoney(), ess)));
+                    sender.sendMessage(tl("setBalOthers", player.getDisplayName(), NumberUtil.displayCurrency(player.getMoney(), ess)));
+                    break;
+                }
             }
-            player.sendMessage(tl("takenFromAccount", NumberUtil.displayCurrency(player.getMoney(), ess)));
-        } else {
-            throw new ChargeException(tl("insufficientFunds"));
-        }
-    }
-
-    private void set(BigDecimal amount, final User player, final CommandSource sender) throws MaxMoneyException {
-        BigDecimal minBalance = ess.getSettings().getMinMoney();
-        BigDecimal maxBalance = ess.getSettings().getMaxMoney();
-        boolean underMinimum = (amount.compareTo(minBalance) < 0);
-        boolean aboveMax = (amount.compareTo(maxBalance) > 0);
-        player.setMoney(underMinimum ? minBalance : aboveMax ? maxBalance : amount, UserBalanceUpdateEvent.Cause.COMMAND_ECO);
-        player.sendMessage(tl("setBal", NumberUtil.displayCurrency(player.getMoney(), ess)));
-        if (sender != null) {
-            sender.sendMessage(tl("setBalOthers", player.getDisplayName(), NumberUtil.displayCurrency(player.getMoney(), ess)));
-        }
+        });
     }
 
 
     private enum EcoCommands {
         GIVE, TAKE, SET, RESET
     }
 
+    @Override
+    protected void updatePlayer(Server server, CommandSource sender, User user, String[] args) throws NotEnoughArgumentsException, PlayerExemptException, ChargeException, MaxMoneyException {
```

Comment:
md wanted this to remain abstract https://github.com/EssentialsX/Essentials/pull/3337#discussion_r432844893

As for consumers vs this method, consumers allows us not to have to use instance/static variables laying around the command.

Reply:
Fair enough. I don't dislike the new method, but it could perhaps be its own distinct class (still abstract), since only one or the other method is ever being used.